compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmValueGen.java

```diff
@@ -461,19 +461,17 @@ private void createObjectGetMethod(ClassWriter cw, Map<String, BField> fields, S
     private void createObjectSetMethod(ClassWriter cw, Map<String, BField> fields, String className,
                                        JvmCastGen jvmCastGen) {
 
-        createObjectSetMethod(cw, fields, className, "set", "checkFieldUpdate", jvmCastGen);
+        createObjectSetMethod(cw, fields, className, "set", jvmCastGen, false);
```

Comment:
instead of passing true or false, should we create two separate methods?

Reply:
I'll Fix this in the master fix. Need to include minor improvements to Typechecker too.